renku/infrastructure/storage/storage_service.py

```diff
@@ -0,0 +1,154 @@
+
+    @cached_property
+    def project_id(self) -> Optional[str]:
+        """Get the current gitlab project id.
+
+        Note: This is mostly a workaround since storage service is already done to only accept
+        project ids, but the CLI knows nothing about those.
+        This could should be removed once we move to proper renku projects.
+        """
+        namespace, name = Project.get_namespace_and_name(
+            remote=project_context.remote, name=project_context.project.name, repository=project_context.repository
+        )
+
+        if namespace is None or name is None:
+            raise errors.ParameterError("Couldn't get namespace or name for current project")
+        if namespace.startswith("repos/"):
+            namespace = namespace[6:]
+        gitlab_url = f"https://{project_context.remote.host}/repos/"
+
+        return self._gl.get_project_id(gitlab_url, namespace, name)
+
+    def _auth_headers(self) -> Dict[str, Any]:
+        """Send a request with authentication headers."""
+        token = read_renku_token(None, get_endpoint_from_remote=True)
+        if not token:
+            raise errors.NotLoggedIn("Must be logged in to get access storage for a project.")
+
+        return {"Authorization": f"Bearer {token}"}
+
+    def _send_request(
+        self,
+        path: str,
+        parameters: Optional[Dict[str, Any]] = None,
+        body: Optional[Dict[str, Any]] = None,
+        method="GET",
+        auth=False,
+        expected_response=[200],
+    ):
+        """Send an unauthenticated request."""
+        request_method: Callable[..., Any]
+        if method == "GET":
+            request_method = requests.get
+        elif method == "POST":
+            request_method = requests.post
+        elif method == "PUT":
+            request_method = requests.put
+        elif method == "DELETE":
+            request_method = requests.delete
+        else:
+            raise NotImplementedError()
+
+        url = f"{self.base_url}{path}"
+        headers = None
+
+        if auth:
+            headers = self._auth_headers()
+
+        resp = request_method(url, headers=headers, params=parameters, data=body, timeout=TIMEOUT)  # type: ignore
+
+        if resp.status_code not in expected_response:
+            raise errors.RequestError(f"Request to storage service failed ({resp.status_code}): {resp.text}")
+
+        return resp.json()
+
+    def list(self, project_id: str) -> List[CloudStorageWithSensitiveFields]:
+        """List storage configured for the current project."""
+        response = self._send_request("/storage", parameters={"project_id": project_id}, auth=True)
+        results = []
+        for res in response:
+            results.append(
+                CloudStorageWithSensitiveFields(CloudStorage.from_dict(res["storage"]), res["sensitive_fields"])
+            )
+
+        return results
+
+    def create(self, storage: CloudStorage) -> CloudStorageWithSensitiveFields:
```

Comment:
As far as I can tell create, edit, delete and validate are currently not being used, right?

Reply:
yes they're only placeholders. originally the build should have included full cli support but that was scoped out. back then we said that this should be done immediately after the build, so i left it in. I guess now that we didn't do that, it could also be removed